packages/twenty-server/src/modules/workflow/common/services/workflow-common.workspace-service.ts

```diff
@@ -40,24 +37,21 @@ export class WorkflowCommonWorkspaceService {
 
   async getValidWorkflowVersionOrFail(
     workflowVersion: WorkflowVersionWorkspaceEntity | null,
-  ): Promise<
-    Omit<WorkflowVersionWorkspaceEntity, 'trigger'> & {
-      trigger: WorkflowTrigger;
-    }
-  > {
+  ): Promise<WorkflowVersionWorkspaceEntity> {
     if (!workflowVersion) {
       throw new WorkflowTriggerException(
         'Workflow version not found',
         WorkflowTriggerExceptionCode.INVALID_INPUT,
       );
     }
 
-    if (!workflowVersion.trigger) {
-      throw new WorkflowTriggerException(
-        'Workflow version does not contains trigger',
-        WorkflowTriggerExceptionCode.INVALID_WORKFLOW_VERSION,
-      );
-    }
+    // FIXME: For now we will make the trigger optional. Later, we'll have to ensure the trigger is defined when publishing the flow.
+    // if (!workflowVersion.trigger) {
+    //   throw new WorkflowTriggerException(
+    //     'Workflow version does not contains trigger',
+    //     WorkflowTriggerExceptionCode.INVALID_WORKFLOW_VERSION,
+    //   );
+    // }
```

Comment:
why trigger can be optional at workflow publication?

Reply:
Sorry, the trigger should not be optional at publication.
The code ensured that the last version was valid every time the workflow was updated. That way, when publishing a workflow, it ensured that the last version was valid and could indeed be published.
However, that doesn't work for the initial version of the workflow. Indeed, the trigger is `null` for the first workflow version and will be provided later by the user.
We must bring back a validity check to workflow versions before publishing. But I don't know how to do it best.